### ISSUE TITLE
Prevent keyboard from hiding when receiving and sending new messages

### ISSUE THREAD

**Issue by classiebit** [FaridSafi/react-native-gifted-chat#2042]
#### Issue Description

The keyboard gets dismissed whenever a new message is received or sending a new message. 

#### Steps to Reproduce / Code Snippets

    <View style={[GS.container, {paddingBottom: useSafeAreaInsets().bottom}]}>
        <TopNavigation
            title={routeUser.name}
            subtitle={'  '+format(new Date(routeUser.time))}
            accessoryLeft={_goBack}
            accessoryRight={_renderProfile}
        />

        <GiftedChat
                messages={messages}
                onSend={message => _onSend(message)}
                user={{ _id: authState.auth.id}}
            />
    </View>

#### Expected Results

Basically, the keyboard is getting dismissed whenever the **messages** state updates, which we don't want.

#### Additional Information

* Nodejs version: 15.11.0
* React version: 16.13.1
* React Native version: 0.63.2
* react-native-gifted-chat version: 0.16.3
* Platform(s) (iOS, Android, or both?): both
* TypeScript version: 4.0.0

**Comment by classiebit** [comment]
But, when we remove the parent components and just use GiftedChat without <View> and <TopNavigation> component, like this-

    <GiftedChat
        messages={messages}
        onSend={message => _onSend(message)}
        user={{ _id: authState.auth.id}}
    />


Then it's working fine. But we need to use it inside the View component according to the User-interface design.

**Comment by evan-klemen-cleo** [comment]
@classiebit did you have any luck on this? I'm running into the same issue where keyboard dismisses on Send.

**Comment by SherrieTsao** [comment]
Same here. Tried to add `textInputAutoFocus={true}` in `<Composer />`. It worked, but it ain't good UX - keyboard dismisses first then re-popup.

**Comment by classiebit** [comment]
@evan-klemen-cleo Yes, we resolved the issue. The issue was not with GiftedChat but with the Parent component. Hence, if you're using GiftedChat within a View component, try to remove Styles of that View component.

**Comment by evan-klemen-cleo** [comment]
@classiebit 
Ive removed all styles from the parent `View` tag other than a height (otherwise nothing shows up on screen, and still no luck. The next parent above that is the stack navigator.

**Comment by SherrieTsao** [comment]
> @evan-klemen-cleo Yes, we resolved the issue. The issue was not with GiftedChat but with the Parent component. Hence, if you're using GiftedChat within a View component, try to remove Styles of that View component.

This makes so much sense bc I put gifted chat in a separate component. 

> @classiebit
> Ive removed all styles from the parent `View` tag other than a height (otherwise nothing shows up on screen, and still no luck. The next parent above that is the stack navigator.

Would you mind posting a snippet? I possibly tried everything and finally worked.

**Comment by classiebit** [comment]
Here's the complete screen code-

(`<Layout>` is from UIKitten Framework- `{ Layout, Icon, Button } from '@ui-kitten/components';`)

     <Layout style={[GS.container, {paddingBottom: useSafeAreaInsets().bottom}]}>
            <Layout style={styles.row}>
                <Button appearance='ghost' accessoryLeft={back} onPress={_goBack} style={styles.backButton}/>
                
                <Avatar style={styles.row__avatar} source={{uri: API_STORAGE+routeUser.avatar}} />
                <Layout style={styles.row__info}>
                    <Text style={styles.row__name}>{routeUser.name}</Text>
                    <Text style={styles.row__position}>{format(new Date(routeUser.time))}</Text>
                </Layout>
                
                <Button appearance='ghost' accessoryLeft={video} onPress={_videoCall} style={styles.backButton}/>
            </Layout>

            <GiftedChat
                messages={messages}
                onSend={message => _onSend(message)}
                user={{
                    _id: authState.auth.id,
                }}
                renderAvatar={null}
                renderSend={_renderSend}
                renderActions={_renderCustomActions}
                scrollToBottom
                keyboardShouldPersistTaps='never'
                parsePatterns={_parsePatterns}
                renderBubble={_renderBubble}
                renderInputToolbar={_renderInputToolbar}
                renderComposer={props => ( <Composer {...props} textInputStyle={{ color: (colorScheme === 'dark' ? Colors.white : Colors.textDark)}} /> )} 
                onInputTextChanged={(text) => _onTyping(text)}
                isTyping={typing}
            />
            
 </Layout>


     const styles = StyleSheet.create({
    profileImage: {
        width: 32,
        height: 32,
        borderRadius: 16,
    },
    
    sendButton: {
        marginRight: 15,
        justifyContent: 'center',
        width: 20,
        height: 20,
    },
    micButton: {
        
    },

    row: {
        flexDirection: 'row',
        width: '100%',
        borderBottomColor: '#b2b2b2',
        borderBottomWidth: .4,
    },
    
    row__avatar: {
        marginTop:8,
        width: 32,
        height: 32,
        borderRadius: 50,
    },
    
    row__info: {
        marginLeft: 12,
    },
    
    row__name: {
        paddingTop: 8,
        fontWeight: '700',
        fontSize: 14,
    },
    
    row__position: {
        color: '#9a9c9d',
        fontSize: 12,
    },

    backButton: {
        width: 12,
        height: 12,
        marginVertical: 4,
    },

    });

**Comment by tungnc3142** [comment]
Same issue, when send message, after keyboard auto hide.

**Comment by mehdinourollah** [comment]
Same issue and I have this re-rendering issue which cause resetting whatever user entered so far !

**Comment by fukemy** [comment]
any solution here?

**Comment by hammad-ahm3d** [comment]
hey guys! were you able to solve this issue?

**Comment by AliSaeed090** [comment]
any solution here?

**Comment by hammad-ahm3d** [comment]
For all those who are having the same issue, I fixed it by following the steps below:

1. Go to the file /android/app/src/main/AndroidManifest.xml and add the following lines inside <activity> tag "android:windowSoftInputMode="stateAlwaysHidden|adjustPan".
2. In package.json, change "react-native-screens" version to "^3.4.0"
3. clean android build folder using, "cd android && ./gradlew clean".
4. remove node_modules, package-lock.json or yarn.lock
5. reinstall node_modules and "react-native run-android" again.

**Comment by JieMin27** [comment]
any other solutions?

**Comment by AliSaeed090** [comment]
I solved this issue by declaring the renderInputToolbar and renderComposer function outside the main functional component.

**Comment by JieMin27** [comment]
Found out that render custom composer cause this issue.
